src/document.rs

```diff
@@ -314,14 +328,18 @@ impl Document {
                 .unwrap_or(default_excerpt_separator);
 
             let excerpt = if let Some(excerpt_str) = excerpt_attr {
-                excerpt_str
+                excerpt_str.to_string()
             } else if excerpt_separator.is_empty() {
-                ""
+                String::new()
             } else {
-                self.content.split(excerpt_separator).next().unwrap_or(&self.content)
+                self.extract_references() +
```

Comment:
I forgot about the potential for liquid posts (or HTML for that matter), so I've fixed that.  Regarding the references before the excerpt (rather than after), that's how I originally had things arranged... but that required changing the type of the returned String from extract references from String to &str (outside the function, because otherwise it doesn't live long enough), changing the type of the extracted excerpt (shown below) to a String, and adding two newlines to separate the references from the excerpt.

    self.content
        .split(excerpt_separator)
        .next()
        .unwrap_or(&self.content)

My original idea was to have the extract references return the type &str, but I don't think I understand the lifetime system well enough, because I was never able to get it to work.

I would like to not have to change the type returned by the "if let" arms in this area: originally it returned &str, and I had to change to Strings to be able to concatenate the extracted references with the excerpt.  Suggestions as to how that would be done are welcome!